querydsl.go

```diff
@@ -1,2 +1,153 @@
 // Package querydsl provides programmatic mapping from the CyVerse search DSL to Elasticsearch queries
 package querydsl
```

Comment:
This package is where a lot of the meat is, and should see the most review.

Reply:
For my rusty "Go" eyes, this looks prettty clean and straight forward.
I don't have much to add for this first pass. The structure looks sane.